trace_decoder/tests/trace_decoder_tests.rs

```diff
@@ -0,0 +1,266 @@
+        .into_iter()
+        .map(|file_path| {
+            {
+                // Read one json witness file for this block and get list of BlockProverInputs
+                read_witness_file(&file_path)
+            }
+        })
+        .map_ok(|block_prover_inputs| {
+            block_prover_inputs.into_iter().map(|block_prover_input| {
+                // Run trace decoder, create list of generation inputs
+                let block_generation_inputs = decode_generation_inputs(block_prover_input)?;
+                block_generation_inputs
+                    .into_par_iter()
+                    .map(|generation_inputs| {
+                        // For every generation input, simulate execution.
+                        // Execution will be simulated in parallel
+                        let timing = TimingTree::new(
+                            &format!(
+                                "Simulating zkEVM CPU for block {} txn {:?}",
+                                generation_inputs.block_metadata.block_number,
+                                generation_inputs.txn_number_before
+                            ),
+                            log::Level::Info,
+                        );
+                        simulate_execution::<F>(generation_inputs)?;
+                        timing.filter(Duration::from_millis(100)).print();
+                        Ok::<(), anyhow::Error>(())
+                    })
+                    .collect::<Result<Vec<_>, anyhow::Error>>()
+            })
+        })
+        .flatten_ok()
+        .map(|it| it?)
+        .collect::<Vec<Result<_, anyhow::Error>>>();
+
+    results.iter().for_each(|it| {
+        if let Err(e) = it {
+            panic!("Failed to run parsing decoding proving test: {e:?}");
+        }
+    });
+}
+
+/// This test checks for the parsing and decoding of the block witness
+/// received from Jerigon and CDK Erigon into zkEVM `GenerationInputs`, and
+/// checks if trace decoder output generation inputs are valid and consistent
+#[rstest]
+#[case(JERIGON_WITNESS_DIR)]
+//#[case(CDK_ERIGON_WITNESS_DIR)]
+fn test_generation_inputs_consistency(#[case] test_witness_directory: &str) {
+    init_logger();
+
+    let result: Vec<Result<(), anyhow::Error>> =
+        find_json_data_files(test_witness_directory, JsonFileType::Witness)
+            .expect("valid json data files found")
+            .into_iter()
+            .map(|file_path| {
+                {
+                    // Read json header file of the block. We need it to check tracer output
+                    // consistency
+                    let mut header_file_path = file_path.clone();
+                    header_file_path.set_extension("");
+                    let mut block_header_file_name = header_file_path
+                        .file_name()
+                        .expect("valid header file name")
+                        .to_os_string();
+                    block_header_file_name.push("_header.json");
```

Comment:
Nonblocking: I'd prefer to have a single document for each vector rather than a naming convention:
```json
{
    "witness": {
        "header": ...,
        "contents": ...
    }
}
```

Reply:
@0xaatif It occur to me we could do it that way, but that would complicate other tools and the witness format we are using by default with `zero-bin` which is now used as test input. Some witness files are very big (10-100MB), so hacking them manually to add header is a bit tedious.

Reply:
Regarding payload sizes, if we were addressing #248, which is trivial, we could reduce them by up to ~15% on some cases, so definitely not negligible.